File: src/main/java/com/rs/java/game/area/banks/WarriorGuildBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class WarriorGuildBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Warrior Guild Bank";
	}

	public WarriorGuildBank() {
		super(
				new Polygon(new WorldTile[] {
				new WorldTile(2843, 3533, 0),
				new WorldTile(2843, 3537, 0),
				new WorldTile(2841, 3537, 0),
				new WorldTile(2841, 3540, 0),
				new WorldTile(2848, 3540, 0),
				new WorldTile(2848, 3533, 0),
				new WorldTile(2843, 3533, 0),
		}) );
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/WarriorGuildBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class WarriorGuildBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2843, 3533, 0),
            WorldTile(2843, 3537, 0),
            WorldTile(2841, 3537, 0),
            WorldTile(2841, 3540, 0),
            WorldTile(2848, 3540, 0),
            WorldTile(2848, 3533, 0),
            WorldTile(2843, 3533, 0),
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Warrior Guild Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
